de.metas.business/src/main/java/org/adempiere/pricing/api/impl/MDiscountSchemaBL.java

```diff
@@ -264,15 +239,11 @@ private boolean hasNoValues(final Collection<I_M_AttributeInstance> instances)
 			return true;
 		}
 
-		for (final I_M_AttributeInstance instance : instances)
-		{
-			if (!hasNoValue(instance))
-			{
-				return false;
-			}
-		}
+		final List<I_M_AttributeInstance> attributeInstances = instances.stream()
+				.filter(instance -> !hasNoValue(instance))
```

Comment:
you can use "anyMatch(instance -> !hasNoValue(instance))" and u can return it directly...
... return it negated as I see below